File: baekjoon/DFS&BFS/5567_결혼식.py
from collections import deque
import sys
input = sys.stdin.readline
n = int(input())
m = int(input())
array = [list(map(int, input().split())) for _ in range(m)]


def bfs(graph):
    cnt = 0  # 결혼식 참석자 수
    queue = deque([])
    visited = [False] * (n+1)
    visited[1] = True
    # 상근이 친구
    for i in graph[1]:
        queue.append(i)
        visited[i] = True
        cnt += 1

    while queue:
        tmp = queue.popleft()
        # 상근이 친구의 친구
        for i in graph[tmp]:
            if visited[i] == False:
                visited[i] = True
                cnt += 1

    return cnt


def solution():
    graph = [[] for _ in range(n+1)]
    for [x, y] in array:
        graph[x].append(y)
        graph[y].append(x)

    return bfs(graph)


print(solution())

File: baekjoon/DFS&BFS/5567_결혼식.js
const fs = require('fs')
let [n, m, ...input] = fs
  .readFileSync('../input.txt')
  .toString()
  .trim()
  .split('\n')
n = Number(n)
m = Number(m)
let array = input.map((el) => el.split(' ').map(Number))

function bfs(graph) {
  let cnt = 0 // 결혼식 참석자 수
  const queue = []
  const visited = Array.from(Array(n + 1).fill(false))
  visited[1] = true
  // 상근이 친구
  for (let i of graph[1]) {
    queue.push(i)
    visited[i] = true
    cnt += 1
  }

  while (queue.length) {
    const tmp = queue.shift()
    // 상근이 친구의 친구
    for (let i of graph[tmp]) {
      if (visited[i] === false) {
        visited[i] = true
        cnt += 1
      }
    }
  }

  return cnt
}

function solution() {
  const graph = Array.from(Array(n + 1), () => [])
  for (const [x, y] of array) {
    graph[x].push(y)
    graph[y].push(x)
  }

  return bfs(graph)
}

console.log(solution())
